src/platform/intel/cavs/lib/clk.c

```diff
@@ -48,6 +53,10 @@ static inline void select_cpu_clock(int freq_idx, bool release_unused)
 		io_reg_write(SHIM_BASE + SHIM_CLKCTL,
 			     (io_reg_read(SHIM_BASE + SHIM_CLKCTL) &
 			      ~SHIM_CLKCTL_OSC_REQUEST_MASK) | enc);
+#if CONFIG_TIGERLAKE
+		if (freq_idx == CPU_LPRO_FREQ_IDX)
+			cpu_disable_core(PLATFORM_CORE_COUNT - 1);
+#endif
```

Comment:
I guess this is on purpose, but looks a bit hard to follow. Wouldn't the correct logic be to disable the count-1 cpu core if any other freq than CPU_HPRO is requested (and not just CPU_LPRO_FREQ_IDX)?

Reply:
well, it doesnt matter because we have only HPRO and LPRO on TGL :)
But I can update with your approach